### DISCUSSION TITLE
Special config vars for plugins?

### DISCUSSION THREAD

**Discussion by zakk4223** [hyprwm/Hyprland#5450] [Q&A]
Is there any way to use the hyprlang 'special' config vars/category in a hyprland plugin? Realized during the easymotion plugin dev having this would be much better than the current config method where the user just stuffs it all into the dispatch arguments.

Also is there a way to declare config variables as some of the Hyprland custom types? Like gradients and gaps? I did it as a string and used the existing parsers but that's kinda bleh

**Comment by vaxerski** [comment]
> Is there any way to use the hyprlang 'special' config vars/category in a hyprland plugin? Realized during the easymotion plugin dev having this would be much better than the current config method where the user just stuffs it all into the dispatch arguments.

once https://github.com/hyprwm/hyprlang/issues/39 lands I'll lift the restriction of the plugins to `plugin {}`

> Also is there a way to declare config variables as some of the Hyprland custom types? Like gradients and gaps? I did it as a string and used the existing parsers but that's kinda bleh

Why not?
